### PR TITLE
Update index.js

### PR DESCRIPTION
**Pull request by starchturrets**
Spent a little over an hour refactoring based on [this reddit comment](https://www.reddit.com/r/learnjavascript/comments/bbb0a3/code_review_looking_for_help_improvingmodifying/ekhuq25/). 

- Added [arrow functions.](https://dev.to/veronicorn/brief-intro-to-arrow-functions-2pc6)
- Replaced a few more lets with const. 
- Simplified the get request with fetch().
- function expressions were replaced with function declarations.
- Removed the multiple variable declarations.
- The code called by the GENERATE_BTN event listener is in its separate function. 

As far as I've tested on localhost, the app still works. Hope this helps!

### CODE REVIEW

**Comment by beepye** [comment]
_Damn_! This is great! Thanks, I _really_ appreciate this. Sorry for the late response, I've been absent for a few days. I'm going to pour over what you did and wrap my head around it before I merge it.

File: js/index.js [hunk 1]
```diff
@@ -1,135 +1,104 @@
...
+            removeList();
+        }
+
+        GENERATE_BTN.textContent = newMessage;
+        count++;
+        if (count % 10 === 0) {
+            GENERATE_BTN.textContent = `Are you effing serious?`;
+        }
+        //Replaced the callbacks (which I don't really understand) and XHR with fetch(). 
+        fetch(URL).then(response => response.json()).then(response => createList(response));
```
Comment (beepye) [inline js/index.js]: _**That**_ is awesome.

Comment (beepye) [inline js/index.js]: `fetch(URL).then(response => response.json()).then(response => createList(response));`
**_That ^_**. That is awesome. The chaining is nice too, I'm guessing that's particular to `fetch`?

File: js/index.js [hunk 2]
```diff
@@ -1,135 +1,104 @@
...
+                const button = item.closest('button');
+                button.addEventListener('click', removeListItem);
+            });
+        }
+    }
+
+
+
+    function selectRandomMeals(days, arr, callback) {
+        const randomlyOrderedArray = arr.sort(() => {
```
Comment (beepye) [inline js/index.js]: `arr.sort(() => {}`
So arrow functions are simply anonymous functions that replace the word `function` with `=>`? Or is there more to it than that?

File: js/index.js [hunk 3]
```diff
@@ -1,135 +1,104 @@
...
+        `Keep on trying`,
+        `I Don't like this` //Template literals preserve whitespace and quotation marks.
+    ];
+
+    let count = 0;
+    GENERATE_BTN.addEventListener('click', (ev) => main(ev));
+
+
+    function main(ev) {
+        const listContainer = document.querySelectorAll('.content-container')[1]; //I tried setting it to query an id, but I kept getting undefined errors for some reason. Leaving it alone for now.
```
Comment (beepye) [inline js/index.js]: No worries here - I updated the selector by adding a specific `js-` prefixed class

File: js/index.js [hunk 4]
```diff
@@ -1,135 +1,104 @@
...
+    const URL = 'http://www.thebrianpye.com/Recipes/meals.json';
+    const BTN_TEXT = [
+        `Make it again`,
+        `Nope, try again`,
+        `Keep on trying`,
+        `I Don't like this` //Template literals preserve whitespace and quotation marks.
+    ];
+
+    let count = 0;
+    GENERATE_BTN.addEventListener('click', (ev) => main(ev));
```
Comment (beepye) [inline js/index.js]: `('click', (ev) => main(ev))`
How come it's not just `('click', main)`?

**Comment by starchturrets** [comment]
> I'm guessing that's particular to fetch?

[Promise chaining](https://javascript.info/promise-chaining) is a thing, though I'll admit I barely understand them. 

>So arrow functions are simply anonymous functions that replace the word function with =>? Or is there more to it than that?

Yes, although there are a [few nuances](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#No_separate_this), as far as I know relating to the `this` keyword, along with methods and constructors. 

>How come it's not just ('click', main)?

Well, TIL I was using event listeners wrong. Thanks for the catch!

**Pull request merged**